src/CDFDocument.php

```diff
@@ -20,16 +20,17 @@ public function getEntities() {
   }
 
   public function getCDFEntity($uuid) {
-    if ($this->hasEntity($uuid)) {
-      return $this->entities[$uuid];
-    }
+    return $this->entities[$uuid] ?? null;
   }
 
   public function setCDFEntities(CDFObject ...$entities) {
-    unset($this->entities);
+    //unset($this->entities);
+    //With unset() if $entities list is empty, hasEntities() will give an error
+    $entitiesList = [];
     foreach ($entities as $entity) {
-      $this->entities[$entity->getUuid()] = $entity;
+        $entitiesList[$entity->getUuid()] = $entity;
```

Comment:
Indentation is wrong, should be 2, not 4. (Drupal is really weird to go with 2 ...)